plugins/services/src/js/components/ServicesQuotaOverviewDetail.tsx

```diff
@@ -0,0 +1,132 @@
+import React from "react";
+import { Trans } from "@lingui/macro";
+
+import { findNestedPropertyInObject } from "#SRC/js/utils/Util";
+import ProgressBar from "#SRC/js/components/ProgressBar";
+import * as ResourcesUtil from "#SRC/js/utils/ResourcesUtil";
+import Loader from "#SRC/js/components/Loader";
+
+import { ServiceGroup, QuotaResources } from "../types/ServiceGroup";
+
+export interface ServicesQuotaOverviewDetailProps {
+  group: ServiceGroup;
+}
+
+function getQuotaPercentage(
+  group: ServiceGroup,
+  resource: string
+): number | null {
+  const resourceQuota: QuotaResources | undefined = findNestedPropertyInObject(
+    group.quota,
+    resource
+  );
+  if (!resourceQuota || !resourceQuota.limit) {
+    return null;
+  }
+
+  if (!resourceQuota.consumed) {
+    return 0;
+  }
+
+  return (resourceQuota.consumed / resourceQuota.limit) * 100;
+}
+
+function getQuotaConsumedOfLimit(
+  group: ServiceGroup,
+  resource: string
+): React.ReactNode | null {
+  const resourceQuota: QuotaResources | undefined = findNestedPropertyInObject(
+    group.quota,
+    resource
+  );
+
+  if (!resourceQuota || !resourceQuota.limit) {
+    return null;
+  }
+
+  if (resource === "cpus" || resource === "gpus") {
+    return (
+      <Trans>
+        {resourceQuota.consumed} of {resourceQuota.limit} Cores
+      </Trans>
+    );
+  }
+  return (
+    <Trans>
+      {resourceQuota.consumed} of {resourceQuota.limit} MiB
+    </Trans>
+  );
+}
+
+class ServicesQuotaOverviewDetail extends React.Component<
+  ServicesQuotaOverviewDetailProps,
+  {}
+> {
+  constructor(props: Readonly<ServicesQuotaOverviewDetailProps>) {
+    super(props);
+
+    this.getCard = this.getCard.bind(this);
+  }
+
+  getCard(resource: string): React.ReactNode {
+    const { group } = this.props;
+
+    const title =
+      resource === "cpus" || resource === "gpus"
+        ? resource.slice(0, -1).toUpperCase()
+        : resource[0].toUpperCase() + resource.slice(1);
```

Comment:
that looks complicated. could we achieve the same result with a dumb lookup?

what would you think of introducing a resource type and a function like the following?
```typescript
const titleFor = (resource: Resource):string => {
  switch (resource) {
    case "cpus": return "CPUs"
    case "gpus": return "GPUs"
    case "disk": return "Disk"
    case "mem": return "Mem"
    // no default here!
  }
}
```

that function would automatically force developers to provide a title for a new resource, if they add one.